lib/tokenize.cpp

```diff
@@ -8296,12 +8296,18 @@ const Token * Tokenizer::findGarbageCode() const
 
     // case keyword must be inside switch
     for (const Token *tok = tokens(); tok; tok = tok->next()) {
-        if (Token::simpleMatch(tok, "switch (") && Token::simpleMatch(tok->linkAt(1), ") {"))
-            tok = tok->linkAt(1)->linkAt(1);
-        else if (tok->str() == "(")
+        if (Token::simpleMatch(tok, "switch (") && Token::simpleMatch(tok->linkAt(1), ")")) {
+            if (tok->linkAt(1)->linkAt(1) && tok->linkAt(1)->linkAt(1)->str() == "}") {
```

Comment:
I would say that `Token::simpleMatch(tok->linkAt(1), ") {")` is enough as condition here. Unless we would expect that the links are wrong. So the code could be:

    if (Token::simpleMatch(tok, "switch (")) {
        if (Token::simpleMatch(tok->linkAt(1), ") {"))
            tok = tok->linkAt(1)->linkAt(1);
        else .. goto semicolon ..
    }

For information, I did not check the end-parenthesis ")" because I doubted it. I just checked it to make the code a bit more readable.

Reply:
Makes good sense :) I have pushed a new commit with this.